src/server/dragonfly_test.cc

```diff
@@ -541,19 +541,19 @@ TEST_F(DflyEngineTest, Bug496) {
         db.RegisterOnChange([&cb_hits](DbIndex, const DbSlice::ChangeReq&) { cb_hits++; });
 
     {
-      auto res = db.AddOrFind({}, "key-1");
+      auto res = std::move(*db.AddOrFind({}, "key-1"));
```

Comment:
Here's a proposal:
Maybe add `V&& operator*() &&;` to `OpResult`, and then we won't have to call `std::move()` on what seems like rvalue types? We do it quite a lot, so I think it's worth it. That's also how `std::optional` allows similar things.